src/mscorlib/src/System/Lazy.cs

```diff
@@ -322,147 +528,16 @@ public T Value
         {
             get
             {
-                Boxed boxed = null;
-                if (m_boxed != null )
+                while (true)
```

Comment:
Having a while loop here will make the Value property to be a bad inlining candidate - bad for steady state perf. Could you please change CreateValue to return the value instead so that it can be `return (_state == null) ? _value : CreateValue();`